pkg/rekor/identity.go

```diff
@@ -72,6 +77,8 @@ type MonitoredValues struct {
 	// Subjects contains a list of subjects that are not specified in a
 	// certificate, such as a SSH key or PGP key email address
 	Subjects []string `yaml:"subjects"`
+	// OIDConstraints contains a list of OIDs and constraints to be checked
+	OIDConstraints []OIDconstraint `yaml:"oidConstraints"`
```

Comment:
I think we should change the struct so that we can match against a list of extension values given a single OID. With the current structure, you would need to specify an OID key-value pair repeatedly to match against multiple instances. So `OIDConstraint` becomes something like:

```
	OID      asn1.ObjectIdentifier  `yaml:"oid"`
	Values []string`yaml:"values"`
```

I would suggest we start with the assumption that all OID values are strings and we support only the Fulcio OIDs.

Note that this wouldn't support a more complex policy that would match against multiple extension values for a given certificate. For example, maybe you want to match all certificates where an repo is X AND a subject is Y. I think this should be tackled separately. This is a useful feature but would require changing all policy checks. We should explore using a policy language too rather than reinventing this.